torchx/lib/torchx/backend.ex

```diff
@@ -1021,41 +1021,85 @@ defmodule Torchx.Backend do
     end)
   end
 
+  @impl true
+  def window_scatter_max(out, tensor, source, init_value, window_dims_tuple, opts) do
+    intermediate_type =
+      tensor.type
+      |> Nx.Type.to_floating()
+      |> to_torch_type()
+
+    unfold_flat = fn tensor ->
+      unfolded =
+        tensor
+        |> unfold_windows(opts[:padding], 0, window_dims_tuple, opts[:strides])
+        |> Torchx.to_nx()
+
+      {to_keep, to_flatten} =
+        unfolded
+        |> Map.get(:shape)
+        |> Tuple.to_list()
+        |> Enum.split(-tuple_size(window_dims_tuple))
+
+      flat_shape =
+        to_keep
+        |> List.to_tuple()
+        |> then(&Tuple.insert_at(&1, tuple_size(&1), Enum.product(to_flatten)))
+
+      Nx.reshape(unfolded, flat_shape)
+    end
+
+    argmax =
+      tensor
+      |> from_nx()
+      |> Torchx.to_type(intermediate_type)
+      |> then(unfold_flat)
+      |> Nx.argmax(axis: -1)
+
+    indices_to_flatten =
+      tensor
+      |> Nx.axes()
+      |> Enum.map(fn axis ->
+        tensor
+        |> Nx.iota(axis: axis)
+        |> then(unfold_flat)
+        |> Nx.take_along_axis(Nx.new_axis(argmax, -1), axis: -1)
```

Comment:
@josevalim This one is kinda tricky due to indexed_add and take_along_axis. Should I refactor it to use Torchx directly instead of going through Nx? The code might become far less readable going that route, though

Reply:
I thought using the torchs `tensor` as a template would allocate it through torchx. I'll look into this and force the backend if it's indeed on the BinaryBackend

Reply:
I don't think it does but you are right. Perhaps it should!

Reply:
@josevalim I think it would be better if we leave allocating the iotas on the same device for another PR.

This ends up touching lots of other functionalities as well.
WDYT?

Reply:
(the code currently has the WIP for this, but I need to hunt a segfault)

Reply:
Fine by me, so please add a TODO and ship it. :)